### PR TITLE
ROC-3738: Do you need more time to respond? not available on due day

### PR DESCRIPTION
**Pull request by rlewan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-3738

### Change description ###

Aligns the backend with changes in https://github.com/hmcts/cmc-citizen-frontend/pull/583

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/rules/MoreTimeRequestRule.java [hunk 1]
```diff
@@ -18,8 +20,14 @@ public void assertMoreTimeCanBeRequested(Claim claim) {
             throw new MoreTimeAlreadyRequestedException("You have already requested more time");
         }
 
-        if (LocalDate.now().isAfter(claim.getResponseDeadline())) {
+        assertIsNotPastDeadline(LocalDateTimeFactory.nowInLocalZone(), claim.getResponseDeadline());
+    }
+
+    void assertIsNotPastDeadline(LocalDateTime now, LocalDate responseDeadline) {
+        LocalDateTime responseDeadlineTime = responseDeadline.atTime(16, 0);
+        if (now.isEqual(responseDeadlineTime) || now.isAfter(responseDeadlineTime)) {
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/rules/MoreTimeRequestRule.java]: isn't isAfter enough?

File: src/test/java/uk/gov/hmcts/cmc/claimstore/rules/MoreTimeRequestRuleTest.java [hunk 2]
```diff
@@ -38,4 +49,38 @@ public void shouldThrowExceptionWhenResponseDeadlinePassed() {
...
+    public void assertIsNotPastDeadlineShouldNotThrowWhenTimeIsBefore4OnDeadlineDay() {
+        LocalDateTime now = LocalDate.now().atTime(15, 59, 59, 999);
+        LocalDate responseDeadline = LocalDate.now();
+        assertThatCode(
+            () -> moreTimeRequestRule.assertIsNotPastDeadline(now, responseDeadline)
+        ).doesNotThrowAnyException();
+    }
+
+    @Test(expected = MoreTimeRequestedAfterDeadlineException.class)
+    public void assertIsNotPastDeadlineShouldThrowWhenTimeIs4OnDeadlineDay() {
```
Comment (timja-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/rules/MoreTimeRequestRuleTest.java]: pm not 0n I assume you meant to have

File: src/main/java/uk/gov/hmcts/cmc/claimstore/rules/MoreTimeRequestRule.java (hunk 1 above)
Comment (rlewan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/rules/MoreTimeRequestRule.java]: If its 4:00 PM it means it's past deadline according to what Marc F. told me. This is aligned with the frontend `dateTime.isSameOrAfter(fourPM(responseDeadline))`

Comment (timja-kainos) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/rules/MoreTimeRequestRule.java]: ah could have sworn he said 4:01 but 4 should be safe enough

File: src/test/java/uk/gov/hmcts/cmc/claimstore/rules/MoreTimeRequestRuleTest.java (hunk 2 above)
Comment (rlewan) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/rules/MoreTimeRequestRuleTest.java]: What I wanted to say was _"... when it's 4 on deadline day"_, can add PM here too.

**Pull request merged**